isolator/network_isolator.cpp

```diff
@@ -51,108 +51,192 @@
+
+  waitpid(child.get().pid(), NULL, 0);
+  string output = process::io::read(child.get().out().get()).get();
+  LOG(INFO) << "Got response: " << output << " from " << path;
+
+  Try<JSON::Object> jsonOutput = JSON::parse<JSON::Object>(output);
+  if (jsonOutput.isError()) {
+    return Error(
+        "Error parsing output '" + output + "' to JSON string" +
+        jsonOutput.error());
+  }
+
+  Try<OutProto> result = protobuf::parse<OutProto>(jsonOutput.get());
+  if (jsonOutput.isError()) {
+    return Error(
+        "Error parsing output '" + output + "' to Protobuf" + result.error());
+  }
+
+  return result;
+}
 
 
 Try<Isolator*> CalicoIsolatorProcess::create(const Parameters& parameters)
 {
-  std::string ipamPath = "";
+  string ipamClientPath;
+  string isolatorClientPath;
   foreach (const Parameter& parameter, parameters.parameter()) {
-    if (parameter.key() == ipamKey) {
-      ipamPath = parameter.value();
+    if (parameter.key() == ipamClientKey) {
+      ipamClientPath = parameter.value();
+    } else if (parameter.key() == isolatorClientKey) {
+      isolatorClientPath = parameter.value();
     }
   }
-  if (ipamPath == "") {
+  if (ipamClientPath.empty()) {
     LOG(WARNING) << "IPAM path not specified";
     return Error("IPAM path not specified.");
   }
   return new CalicoIsolator(process::Owned<CalicoIsolatorProcess>(
-      new CalicoIsolatorProcess(ipamPath, parameters)));
+      new CalicoIsolatorProcess(
+          ipamClientPath, isolatorClientPath, parameters)));
+}
+
+
+CalicoIsolatorProcess::CalicoIsolatorProcess(
+    const std::string& ipamClientPath_,
+    const std::string& isolatorClientPath_,
+    const Parameters& parameters_)
+  : ipamClientPath(ipamClientPath_),
+    isolatorClientPath(isolatorClientPath_),
+    parameters(parameters_)
+{
+  Try<string> result = net::getHostname(self().address.ip);
+  if (result.isError()) {
+    LOG(FATAL) << "Failed to get hostname: " << result.error();
+  }
+  hostname = result.get();
 }
 
 
 process::Future<Option<ContainerPrepareInfo>> CalicoIsolatorProcess::prepare(
     const ContainerID& containerId,
     const ExecutorInfo& executorInfo,
-    const std::string& directory,
-    const Option<std::string>& rootfs,
-    const Option<std::string>& user)
+    const string& directory,
+    const Option<string>& rootfs,
+    const Option<string>& user)
 {
   LOG(INFO) << "CalicoIsolator::prepare";
 
-  std::string ipAddress = "auto";
-  std::string profile = "none";
-  foreach (const Environment_Variable& var,
-      executorInfo.command().environment().variables()) {
-    LOG(INFO) << "ENV: " << var.name() << "=" << var.value();
-    if (var.name() == "CALICO_IP") {
-      ipAddress = var.value();
-    }
-    else if (var.name() == "CALICO_PROFILE") {
-      profile = var.value();
-    }
+  if (!executorNetgroups->contains(executorInfo.executor_id())) {
+    return Failure(
+        "netgroup label not found for executor: " +
```

Comment:
So this builds and runs fine, but sometime after launching an app on the cluster, I hit this error:
```
E0811 22:26:55.984792   220 slave.cpp:3314] Container '26cc1aca-322d-4802-a820-c5fc337580c2' for executor 'sample-flask-app-2.12bc469c-4078-11e5-96de-0242ac11003f' of framework '20150811-222619-1023414700-5050-1-0000' failed to start: netgroup label not found for executor: sample-flask-app-2.12bc469c-4078-11e5-96de-0242ac11003f
```

Do I need to add something to my [yaml file](https://github.com/djosborne/metaswitch-modules/blob/test-ipam-isolator-interface/test/sample-flask-app-2.json)?